contracts/interchain-token/src/contract.rs

```diff
@@ -1,30 +1,220 @@
-use soroban_sdk::{contract, contractimpl, Address, Env};
+use axelar_soroban_std::ensure;
+use soroban_sdk::token::{self, Interface as _};
+use soroban_sdk::{contract, contractimpl, Address, Bytes, Env, IntoVal, String};
+use soroban_token_sdk::metadata::TokenMetadata;
+use soroban_token_sdk::TokenUtils;
 
+use crate::error::ContractError;
 use crate::event;
 use crate::storage_types::DataKey;
+use crate::utils::{
+    admin, check_nonnegative_amount, extend_instance_ttl, read_allowance, read_balance,
+    read_decimal, read_name, read_symbol, receive_balance, spend_allowance, spend_balance,
+    write_allowance, write_metadata,
+};
 
 #[contract]
 pub struct InterchainToken;
 
 #[contractimpl]
 impl InterchainToken {
-    pub fn __constructor(env: Env, owner: Address) {
-        env.storage().instance().set(&DataKey::Owner, &owner);
+    pub fn __constructor(
+        env: Env,
+        interchain_token_service: Address,
+        admin: Address,
+        minter: Address,
+        token_id: Bytes,
+        token_meta_data: TokenMetadata,
+    ) -> Result<(), ContractError> {
+        ensure!(!token_id.is_empty(), ContractError::TokenIdZero);
+
+        Self::validate_token_metadata(token_meta_data.clone())?;
+
+        env.storage().instance().set(&DataKey::TokenId, &token_id);
+
+        write_metadata(&env, token_meta_data);
+
+        env.storage().instance().set(&DataKey::Admin, &admin);
+        env.storage()
+            .persistent()
+            .set(&DataKey::Minter(minter), &true);
```

Comment:
```suggestion
            .set(&DataKey::Minter(minter), &());
```
set unit value, use storage `.has` and `.remove` to check and remove minters